cpp/src/arrow/flight/client.cc

```diff
@@ -108,7 +108,7 @@ class FinishableStream {
   std::shared_ptr<Stream> stream() const { return stream_; }
 
   /// \brief Finish the call, adding server context to the given status.
-  virtual Status Finish(Status&& st) {
+  virtual Status Finish(Status st) {
```

Comment:
This PR shouldn't touch this code unless it's failing at building?

Reply:
This is necessary after [changing Result::status to return const reference](https://github.com/apache/arrow/pull/7156/files/ebfd82800627d66f22ebf994d84dc76ce6b7465d#diff-45528b85a090fba5be3b1d27d9d2cd1d); const ref can't be converted to rvalue ref